src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -174,8 +174,10 @@ ApplicationManagerImpl::ApplicationManagerImpl(
     , policy_handler_(new policy::PolicyHandler(policy_settings, *this))
     , protocol_handler_(NULL)
     , request_ctrl_(am_settings)
-    , hmi_so_factory_(NULL)
-    , mobile_so_factory_(NULL)
+    , hmi_so_factory_(hmi_apis::HMI_API())
+    , mobile_so_factory_(mobile_apis::MOBILE_API())
+    , v4_protocol_so_factory_(
+          ns_smart_device_link_rpc::V1::v4_protocol_v1_2_no_extra())
```

Comment:
```suggestion
```

these lines are implied, I think we should remove them

Reply:
Done in 4c89818